src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -896,27 +903,26 @@ internal virtual void ValidateParameters()
             }
 
             // Output ??
-            if (PassThru.IsPresent && OutFile is null)
+            if (OutFile is not null && OutFolder is not null)
             {
-                ErrorRecord error = GetValidationError(WebCmdletStrings.OutFileMissing, "WebCmdletOutFileMissingException", nameof(PassThru));
+                ErrorRecord error = GetValidationError(WebCmdletStrings.OutputConflict, "WebCmdletOutputConflictException");
                 ThrowTerminatingError(error);
             }
 
-            // Resume requires OutFile.
-            if (Resume.IsPresent && OutFile is null)
+            if (PassThru.IsPresent && OutFile is null && OutFolder is null)
             {
-                ErrorRecord error = GetValidationError(WebCmdletStrings.OutFileMissing, "WebCmdletOutFileMissingException", nameof(Resume));
+                ErrorRecord error = GetValidationError(WebCmdletStrings.PassThruOutFileMissing, "WebCmdletPassThruOutFileMissingException");
                 ThrowTerminatingError(error);
             }
 
-            _qualifiedOutFile = ShouldSaveToOutFile ? QualifiedOutFile : null;
-
-            // OutFile must not be a directory to use Resume.
-            if (Resume.IsPresent && Directory.Exists(_qualifiedOutFile))
+            // Resume requires OutFile and can't be used with OutFolder..
+            if (Resume.IsPresent && OutFile is null)
```

Comment:
Why Resume can't be used with OutFolder? I'd expect we get the same file name from the request.

And typo.
```suggestion
            // Resume requires OutFile and can't be used with OutFolder.
            if (Resume.IsPresent && OutFile is null)
```

Reply:
It can't be used because the filename for OutFolder is calculated after the response and Resume is before the request

Reply:
Maybe not for the PR. We could make first request with range 0-0 (first byte) and see if we got the filename. It is not expensive.